tests/store/test_hdfs_artifact_repo.py

```diff
@@ -0,0 +1,134 @@
+# pylint: disable=redefined-outer-name
```

Comment:
It would be good if the Travis test environment actually had Hadoop in order to test this, since otherwise this code will just rot. Any way to install it in there? Does Hadoop or pyarrow come with some kind of mini HDFS cluster that one can use for tests? Or alternatively, can we mock stuff in either library to still test the logic here?

Reply:
We have tried with docker image  sequenceiq/hadoop-docker:2.7.0 to verify this.

Reply:
There is no python client available with hadoop mini cluster. Pyarrow too uses the hadoop connection itself. Let me know,if we can go with docker image.

Reply:
How do PyArrow's own unit tests work? There's got to be a way to mock out Hadoop without actually running it. We can use standard Python mocking libraries too, like we did with the various cloud artifact stores (S3, GCS, etc).